### PR TITLE
Add support for EVENT messages 

### PR DESCRIPTION
**Pull request by Vicidel**
Hello all, working with PX4 we are starting to use EVENT instead of STATUS_TEXT for more details over the status of the autopilot. For this we needed to have the EVENT messages bridged to ROS2. This is my implementation, taking input from other people to have it merged in upstream mavros

### CODE REVIEW

File: mavros/src/plugins/sys_status.cpp [hunk 1]
```diff
@@ -738,6 +743,55 @@ class SystemStatusPlugin : public plugin::Plugin
     }
   }
 
+  /**
+   * Sent EVENT message to rosout
+   *
+   * @param[in] severity
+   */
+  void process_event_normal(uint8_t severity, uint32_t px4_id, const std::array<uint8_t, 40>& arguments)
```
Comment (vooon) [inline mavros/src/plugins/sys_status.cpp:751]: As i remember message class has type definition for arrays, better to use it.

File: mavros_msgs/CMakeLists.txt [hunk 2]
```diff
@@ -83,6 +83,7 @@ set(msg_files
   msg/RTKBaseline.msg
   msg/RadioStatus.msg
   msg/State.msg
+  msg/StatusEvent.msg
```
Comment (vooon) [inline mavros_msgs/CMakeLists.txt:86]: That list should be generated by cog.

File: mavros/src/plugins/sys_status.cpp [hunk 3]
```diff
@@ -946,6 +1000,26 @@ class SystemStatusPlugin : public plugin::Plugin
...
+  {
+    uint8_t severity = eventm.log_levels & 0x0F; // take only 4 LSB, see https://mavlink.io/en/messages/common.html#EVENT
+    uint32_t px4_id = eventm.id & 0x00FFFFFF; // take only 24 LSB, see PX4 "constexpr uint32_t ID(const char (&name)[N])" function
+    process_event_normal(severity, px4_id, eventm.arguments);
+
+    auto evt_msg = mavros_msgs::msg::StatusEvent();
+    evt_msg.header.stamp = node->now();
+    evt_msg.severity = severity;
+    evt_msg.px4_id = px4_id;
+    evt_msg.event_time_boot_ms = eventm.event_time_boot_ms;
```
Comment (vooon) [inline mavros/src/plugins/sys_status.cpp:1016]: Usually time boot used to calculate header's time stamp.

Comment (Vicidel) [reply mavros/src/plugins/sys_status.cpp]: Done in new commit

File: mavros_msgs/CMakeLists.txt (hunk 2 above)
Comment (Vicidel) [reply mavros_msgs/CMakeLists.txt:86]: Is there any readme on how to do so? I installed cog, but running the `mavros_cog.py` script does not do anything...

File: mavros/src/plugins/sys_status.cpp (hunk 1 above)
Comment (Vicidel) [reply mavros/src/plugins/sys_status.cpp:751]: From what I see in `mavlink_msg_event.hpp`, they use `std::array<uint8_t, 40> arguments;` the same way. Anything I should change here?

File: mavros_msgs/CMakeLists.txt (hunk 2 above)
Comment (Vicidel) [reply mavros_msgs/CMakeLists.txt:86]: ok thanks, should be good now :ok_hand:

File: mavros/src/plugins/sys_status.cpp [hunk 4]
```diff
@@ -1010,10 +1010,10 @@ class SystemStatusPlugin : public plugin::Plugin
     process_event_normal(severity, px4_id, eventm.arguments);
 
     auto evt_msg = mavros_msgs::msg::StatusEvent();
-    evt_msg.header.stamp = node->now();
+    evt_msg.header.stamp.sec = event_time_boot_ms / 1000;
+    evt_msg.header.stamp.nanosec = (event_time_boot_ms % 1000) * 1000000;
```
[the remark this answers is not in the document]
Comment (Vicidel) [reply mavros/src/plugins/sys_status.cpp]: oh that's nice, added a commit for this

File: mavros/src/plugins/sys_status.cpp (hunk 1 above)
Comment (vooon) [reply mavros/src/plugins/sys_status.cpp:751]: That's okay.

**Review by vooon — APPROVED** [review]
LGTM, Thanks!

**Comment by Vicidel** [comment]
Perfect, thank you!

**Pull request merged**